org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/internal/converter/ZigBeeConverterSwitchLevel.java

```diff
@@ -61,10 +72,36 @@ public boolean initializeConverter() {
                 pollingPeriod = POLLING_PERIOD_HIGH;
             }
         } catch (InterruptedException | ExecutionException e) {
-            logger.error("{}: Exception setting reporting ", endpoint.getIeeeAddress(), e);
+            logger.error("{}: Exception setting level control reporting ", endpoint.getIeeeAddress(), e);
         }
 
-        // Add a listener, then request the status
+        clusterOnOffServer = (ZclOnOffCluster) endpoint.getInputCluster(ZclOnOffCluster.CLUSTER_ID);
+        if (clusterOnOffServer != null) {
+            try {
+                CommandResult bindResponse = clusterOnOffServer.bind().get();
+                if (bindResponse.isSuccess()) {
+                    // Configure reporting
+                    CommandResult reportingResponse = clusterOnOffServer
+                            .setOnOffReporting(1, REPORTING_PERIOD_DEFAULT_MAX).get();
+                    if (reportingResponse.isError()) {
+                        pollingPeriod = POLLING_PERIOD_HIGH;
+                    }
+                } else {
+                    pollingPeriod = POLLING_PERIOD_HIGH;
+                }
+            } catch (InterruptedException | ExecutionException e) {
+                logger.error("{}: Exception setting on off reporting ", endpoint.getIeeeAddress(), e);
+            }
+
+            // Set the currentState to ON. This will ensure that we only ignore levelControl reports AFTER we have
+            // really received an OFF report, thus confirming ON_OFF reporting is working
+            currentState.set(true);
+
+            // Add a listener, then request the status
```

Comment:
Is the second part of the comment outdated ("then request the status")?

Reply:
Converters should request the state - it is done systematically elsewhere, so the comment is out of date.